meson.build

```diff
@@ -0,0 +1,43 @@
+project('tinc', 'c',
+  version: run_command('./src/git_tag.sh', '--abbrev=0').stdout().strip(),
+  meson_version: '>=0.49',
+  default_options: [
+    'c_std=c11',
+    'warning_level=3',
+  ],
+)
+
+cc = meson.get_compiler('c')
+os_name = host_machine.system()
+
+cc_flags = ['-D_GNU_SOURCE']
+add_project_arguments(cc_flags, language: 'c')
+
+build_root = meson.build_root()
+src_root = meson.source_root()
+
+prefix = get_option('prefix')
+bindir = prefix / get_option('bindir')
+mandir = prefix / get_option('mandir')
+libdir = prefix / get_option('libdir')
+datadir = prefix / get_option('datadir')
+infodir = prefix / get_option('infodir')
+sbindir = prefix / get_option('sbindir')
+localedir = prefix / get_option('localedir')
+sysconfdir = prefix / get_option('sysconfdir')
+localstatedir = prefix / get_option('localstatedir')
```

Comment:
You can't do this, some directories will be absolute and not live inside prefix, by default that's `syconfdir`, `localstatedir` and `sharedstatedir`. I think you should use [`join_path()`](https://mesonbuild.com/Reference-manual_functions.html#join_paths) here, like so:

```
foodir = join_path(prefix, get_option('foodir')
```

Reply:
I believe it's the same thing?

> (since 0.49.0) Using the / operator on strings is equivalent to calling join_paths().

> If any one of the individual segments is an absolute path, all segments before it are dropped.

I was working from [this](https://gitlab.freedesktop.org/pipewire/pipewire/-/blob/b868ec36c5d16c3d89b1c2005a9f0b9b24a7d415/meson.build#L31-41) and [this](https://gitlab.freedesktop.org/pulseaudio/pulseaudio/-/blob/852c15954ee2c02b8a82cf98f2996adedf6ca0b8/meson.build#L60-73).

Reply:
Ah, I missed that. Then it's fine!